### ISSUE TITLE
Calling "toArray" on array when using array query

### ISSUE THREAD

**Issue by kevin-verschaeve** [ruflin/Elastica#1739]
Hello,

I build a query via a raw php array, and when I request with : 

```
$client->getIndex('index')
	->getType('type')
	->search($arrayQuery)
	->getResults()
;
```

I am getting this error :

```
{
  "type": "https:\/\/tools.ietf.org\/html\/rfc2616#section-10",
  "title": "An error occurred",
  "detail": "Call to a member function toArray() on array",
  "trace": [
    {
      "namespace": "",
      "short_class": "",
      "class": "",
      "type": "",
      "function": "",
      "file": "\/var\/www\/html\/vendor\/ruflin\/elastica\/lib\/Elastica\/Query.php",
      "line": 333,
      "args": []
    },
    {
      "namespace": "Elastica",
      "short_class": "Query",
      "class": "Elastica\\Query",
      "type": "->",
      "function": "toArray",
      "file": "\/var\/www\/html\/vendor\/ruflin\/elastica\/lib\/Elastica\/Search.php",
      "line": 456,
      "args": []
    },
    ...
   ]
}
```

The failing line is :

`if (isset($this->_params['post_filter']) && 0 === count(($this->_params['post_filter'])->toArray())) {`

Removing this line fix the issue (in this case).

I am not sure on how to fix this, and if this is really a bug or a misuse from me.

If this is a bug, I can do a PR checking for the `post_filter` being an array already or not.

Thanks, and happy new year :)

**Comment by ruflin** [comment]
@kevin-verschaeve Which version of Elastica are you using? Are you on master? We did recently quite a few type changes, perhaps @thePanz can chime in here?

**Comment by kevin-verschaeve** [comment]
I am on `6.1.1`

**Comment by ruflin** [comment]
@kevin-verschaeve This is Elastica or Elasticsearch version? Could you share both?

**Comment by kevin-verschaeve** [comment]
Sorry, `elastica 6.1.1`, `elasticsearch 5.0.2`

**Comment by ruflin** [comment]
There is a mismatch between the compatible versions here but my guess is that this isn't the issue. @thePanz any ideas?

**Comment by kevin-verschaeve** [comment]
Everything works fine when I request this way (this is my current workaround actually):

```
$client->request('myindex/mytype/_search', 'GET', $myArrayQuery);
```

For more informations, please be aware that I only run into this issue with the `post_filter` param. Everything else work as expected.

So, I really think the issue is into `Elastica\Query` at line 333

**Comment by ruflin** [comment]
Any chance you could share a var dump of `$this->_params` before the line in question to see what we have in there?

**Comment by kevin-verschaeve** [comment]
Of course, here is an example of a failing request:

```
array(5) {
  ["query"]=>
  array(1) {
    ["bool"]=>
    array(1) {
      ["must"]=>
      array(1) {
        [0]=>
        array(1) {
          ["bool"]=>
          array(1) {
            ["should"]=>
            array(1) {
              [0]=>
              array(1) {
                ["bool"]=>
                array(1) {
                  ["must"]=>
                  array(5) {
                    [0]=>
                    array(1) {
                      ["term"]=>
                      array(1) {
                        ["forSale"]=>
                        array(1) {
                          ["value"]=>
                          bool(true)
                        }
                      }
                    }
                    [1]=>
                    array(1) {
                      ["term"]=>
                      array(1) {
                        ["readyForSale"]=>
                        array(1) {
                          ["value"]=>
                          bool(true)
                        }
                      }
                    }
                    [2]=>
                    array(1) {
                      ["bool"]=>
                      array(1) {
                        ["must"]=>
                        array(2) {
                          [0]=>
                          array(1) {
                            ["exists"]=>
                            array(1) {
                              ["field"]=>
                              string(8) "pictures"
                            }
                          }
                        }
                      }
                    }
                    [3]=>
                    array(1) {
                      ["terms"]=>
                      array(1) {
                        ["brand"]=>
                        array(1) {
                          [0]=>
                          string(11) "XXXX"
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
  ["post_filter"]=>
  array(1) {
    ["bool"]=>
    array(1) {
      ["must"]=>
      array(1) {
        [0]=>
        array(1) {
          ["terms"]=>
          array(1) {
            ["categories.id"]=>
            array(1) {
              [0]=>
              int(1234)
            }
          }
        }
      }
    }
  }
  array(1) {
   ["id"]=>
    string(3) "asc"
  }
  ["size"]=>
  int(3)
}
```

**Comment by ruflin** [comment]
@kevin-verschaeve Sorry I dropped the ball on this one. I wonder if @thePanz has a cycle to check this?

**Comment by thePanz** [comment]
I dropped the ball to @ruflin , the issue is more widespread than this case.
The problem is in how the queries can be created in the elastica library: both as objects, arrays or strings.

This is just one case where we pass an array, but in our code we expect to have an object.
Another case is the bulk-requests, where the data we send to ES is not an object, but an array (and we expect to have an object there). I can not be more specific about this one, as I lost track of my digging down into the rabbit-hole. :(

**Comment by deguif** [comment]
Closing as it was fixed in #1950
